lib/g10_certification_test_kit/bulk_export_validation_tester.rb

```diff
@@ -0,0 +1,170 @@
+require_relative 'profile_guesser'
+
+module BulkExportValidationTester
+  include USCore::MustSupportTest
+  include ProfileGuesser
+
+  MAX_NUM_COLLECTED_LINES = 100
+  MIN_RESOURCE_COUNT = 2
+
+  def observation_metadata
+    [USCore::PediatricBmiForAgeGroup.metadata, USCore::PediatricWeightForHeightGroup.metadata,
+     USCore::ObservationLabGroup.metadata, USCore::PulseOximetryGroup.metadata, USCore::SmokingstatusGroup.metadata,
+     USCore::HeadCircumferenceGroup.metadata, USCore::BpGroup.metadata, USCore::BodyheightGroup.metadata,
+     USCore::BodytempGroup.metadata, USCore::BodyweightGroup.metadata, USCore::HeartrateGroup.metadata,
+     USCore::ResprateGroup.metadata]
+  end
+
+  def diagnostic_metadata
+    [USCore::DiagnosticReportLabGroup.metadata, USCore::DiagnosticReportNoteGroup.metadata]
+  end
+
+  def determine_metadata
+    return observation_metadata if resource_type == 'Observation'
+    return diagnostic_metadata if resource_type == 'DiagnosticReport'
+
+    if resource_type == 'Location' || resource_type == 'Medication'
+      return Array.wrap(USCore::USCoreTestSuite.metadata.find do |meta|
+                          meta.resource == resource_type
+                        end)
+    end
+    ["USCore::#{resource_type}Group".constantize.metadata]
+  end
+
+  def metadata_list
+    @metadata_list ||= determine_metadata
+  end
+
+  def metadata
+    scratch[:metadata] ||= []
```

Comment:
No need for scratch. This could just be an `attr_reader` for a `@metadata` instance variable.